File: my_solution.js
function firstKTimes(array, k) {
   if(!array || !array.length || !k) return -1
    
    const counts = {}
    for(let i = 0; i < array.length; i++){
        // increase the count and keep going
        counts[array[i]] ? counts[array[i]]++ : counts[array[i]] = 1
        // if the count equals k, return the value
        if(counts[array[i]] === k) return array[i];
    }
    
    return -1
}

// Test Cases
console.log(firstKTimes([1, 2, 2, 3, 3], 2)) // 2
console.log(firstKTimes([1, 2, 2, 3, 3], 3)) // -1
console.log(firstKTimes([], 1)) // -1

File: my_solution.py
def firstKTimes(array: [int], k: int) -> int:
    if not array or not len(array) or not k:
        return -1
    
    count = {}
    for num in array:
        if num in count.keys():
            count[num]+=1
        else:
            count[num] = 1
        if count[num] == k:
            return num
    
    return -1

# Test Cases
print(firstKTimes([1, 2, 2, 3, 3], 2)) # 2
print(firstKTimes([1, 2, 2, 3, 3], 3)) # -1
print(firstKTimes([], 1)) # -1
